packages/components/Checkbox/src/Checkbox.tsx

```diff
@@ -0,0 +1,122 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { View, Text } from 'react-native';
+import { ICheckboxState, ICheckboxProps, ICheckboxSlotProps, ICheckboxRenderData, ICheckboxType, checkboxName } from './Checkbox.types';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { filterViewProps } from '@fluentui-react-native/adapters';
+import { settings } from './Checkbox.settings';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { foregroundColorTokens, textTokens, borderTokens } from '@fluentui-react-native/tokens';
+import { useAsPressable, useAsToggleCheckbox } from '@fluentui-react-native/interactive-hooks';
+import { IKeyboardEvent } from '@office-iss/react-native-win32';
+import { backgroundColorTokens } from '@fluentui-react-native/tokens';
+
+export const Checkbox = compose<ICheckboxType>({
+  displayName: checkboxName,
+
+  usePrepareProps: (userProps: ICheckboxProps, useStyling: IUseComposeStyling<ICheckboxType>) => {
+    const { ariaLabel, checked, defaultChecked, boxSide, disabled, label, ...rest } = userProps;
+
+    // Used for uncontrolled Checkbox's to keep internal state
+    const data = useAsToggleCheckbox(defaultChecked || false);
+
+    // On press of a checkbox, call state hook and call client's onChange()
+    const onToggle = React.useCallback(() => {
+      data.onChange();
+      userProps.onChange && userProps.onChange(!data.checked);
+    }, [data, userProps]);
```

Comment:
userProps.onChange might be a slightly better dependency than all of userProps

Reply:
I got rid of onToggle as a whole, check next iteration